deltacat/compute/converter/model/convert_input_files.py

```diff
@@ -21,41 +24,52 @@ def of(
         return result
 
     @property
-    def partition_value(self):
+    def partition_value(self) -> Any:
         return self["partition_value"]
 
     @property
-    def all_data_files_for_dedupe(self):
+    def all_data_files_for_dedupe(self) -> Optional[List[Tuple[int, DataFile]]]:
         return self["all_data_files_for_dedupe"]
 
     @property
-    def applicable_data_files(self):
+    def applicable_data_files(self) -> Optional[List[List[Tuple[int, DataFile]]]]:
         return self["applicable_data_files"]
 
     @property
-    def applicable_equality_delete_files(self):
+    def applicable_equality_delete_files(
+        self,
+    ) -> Optional[List[List[Tuple[int, DataFile]]]]:
         return self["applicable_equality_delete_files"]
 
     @property
-    def existing_position_delete_files(self):
+    def existing_position_delete_files(self) -> Optional[List[Tuple[int, DataFile]]]:
         return self["existing_position_delete_files"]
 
     @partition_value.setter
-    def partition_value(self, partition_value):
+    def partition_value(self, partition_value: Any) -> None:
         self["partition_value"] = partition_value
 
     @all_data_files_for_dedupe.setter
-    def all_data_files_for_dedupe(self, all_data_files_for_dedupe):
+    def all_data_files_for_dedupe(
+        self, all_data_files_for_dedupe: Optional[List[Tuple[int, DataFile]]]
+    ) -> None:
         self["all_data_files_for_dedupe"] = all_data_files_for_dedupe
 
     @applicable_data_files.setter
-    def applicable_data_files(self, applicable_data_files):
+    def applicable_data_files(
+        self, applicable_data_files: Optional[List[List[Tuple[int, DataFile]]]]
```

Comment:
I think some of the type hints like this are a good indicator that we might want to introduce some higher-level structures here to help with code maintainability going forward. ;)

Reply:
Agreed, already got suggested changes :) and will publish in another PR for separation.